From a553c3e3968ed328b341d32e7c627608c6fbb2c6 Mon Sep 17 00:00:00 2001
From: Kees Cook <kees@debian.org>
Date: Wed, 28 Dec 2011 15:22:55 -0800
Subject: [PATCH 1/3] docs: clarify the relationship between relro/bindnow

Clarify the documentation about how bindnow will be forced off if relro
is not enabled or available.

Signed-off-by: Kees Cook <kees@debian.org>
---
 man/dpkg-buildflags.1 |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/man/dpkg-buildflags.1 b/man/dpkg-buildflags.1
index a018edb..b86ae0d 100644
--- a/man/dpkg-buildflags.1
+++ b/man/dpkg-buildflags.1
@@ -231,7 +231,8 @@ This setting (enabled by default) adds
 to \fBLDFLAGS\fP.  During program load, several ELF memory sections need
 to be written to by the linker. This flags the loader to turn these
 sections read-only before turning over control to the program. Most
-notably this prevents GOT overwrite attacks.
+notably this prevents GOT overwrite attacks. If this option is disabled,
+\fBbindnow\fP will become disabled as well.
 .
 .TP
 .B bindnow
@@ -239,7 +240,7 @@ This setting (disabled by default) adds
 .B \-Wl,\-z,now
 to \fBLDFLAGS\fP. During program load, all dynamic symbols are resolved,
 allowing for the entire PLT to be marked read-only (due to \fBrelro\fP
-above).
+above). The option cannot become enabled if \fBrelro\fP is not enabled.
 .
 .TP
 .B pie
-- 
1.7.5.4

